certora/specs/SocialRecoveryModule.spec

```diff
@@ -1,26 +1,160 @@
 using SafeHarness as safeContract;
+using GuardianStorageHarness as guardianStorageContract;
+
+definition SENTINEL() returns address = 1;
 
 methods {
+    // Social Recovery Module Functions
+    function isGuardian(address, address) external returns (bool) envfree;
+    function guardiansCount(address) external returns (uint256) envfree;
+    function threshold(address) external returns (uint256) envfree;
+
+    // Guardian Storage Functions
+    function guardianStorageContract.countGuardians(address) external returns (uint256) envfree;
+
     // Safe Functions
-    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+    function safeContract.isModuleEnabled(address module) external returns (bool) envfree;
+    function safeContract.isOwner(address owner) external returns (bool) envfree;
+
+    // Wildcard Functions (Because of use of ISafe interface in Social Recovery Module)
+    function _.isModuleEnabled(address module) external => safeIsModuleEnabled(calledContract, module) expect bool ALL; // `calledContract` is a special variable.
+    function _.isOwner(address owner) external => sumarizeSafeIsOwner(calledContract, owner) expect bool ALL;
+}
+
+// A summary function to check if a module is enabled in the Safe contract.
+function safeIsModuleEnabled(address callee, address module) returns bool {
+    assert callee == safeContract;
+    return safeContract.isModuleEnabled(module);
 }
 
-// A setup function that requires Safe contract to enabled the Social Recovery
-// Module.
+// A summary function to check if an address is an owner in the Safe contract.
+function sumarizeSafeIsOwner(address callee, address owner) returns bool {
+    assert callee == safeContract;
+    return safeContract.isOwner(owner);
+}
+
+// A setup function that requires Safe contract to enabled the Social Recovery Module.
 function requireSocialRecoveryModuleEnabled() {
     require(safeContract.isModuleEnabled(currentContract));
 }
 
-// This is a dummy rule to verify the Safe Contract Setup with the Social Recovery
-// Module is working as intended.
-rule recoveryModuleCanBeDisabled {
-    env e;
-    address prevModule;
+// This integrity rule verifies that the addGuardianWithThreshold(...) if executes ensures that
+// the Social Recovery Module is enabled, the caller to the Module has to be the Safe Contract,
+// the new guardian is added to the guardian list, and no other account (guardian or not) is affected.
+rule addGuardianWorksAsExpected(env e, address guardian, uint256 threshold, address otherAccount) {
+
```

Comment:
Ah, this appears to be intentional. Personally, not fond of it, but it is a taste thing, so if you think it makes the code easier to read then leave it in.

Reply:
Removed 👍🏾

Reply:
Since you and @akshay-ap are working on the code, I suggest you leave in the extra newline if you and @akshay-ap prefer it. I don't want to impose any style rules here (especially since there is no linter/formatter), as long as there is some consistency (which there was).

Reply:
@akshay-ap any preference? I don't have a strong opinion on this one.